proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/HubResponseResource.java

```diff
@@ -72,4 +85,41 @@ public View hubResponse(
             sessionData.getEidasRelayState()
         );
     }
+
+    private void checkClaimsMatchGatewaySessionData(HttpServletRequest request, GatewaySessionData sessionData) {
+        try {
+            Map<String, Object> claims = sessionCookieService.getData(request.getCookies());
+            if (!sessionDataMatches(sessionData, claims)) {
+                ProxyNodeLogger.warning("sessionData and sessionDataFromCookie are not equal");
```

Comment:
Might be helpful to log out the diff between the two?

Reply:
Only if the data they contain isn't/can't be sensitive

Reply:
It's not, just some request id's, a relay state and an issuer.